### PR TITLE
Coverity: 1381603 1381604 fix to traffic_layout on engine and file_system

### PR DESCRIPTION
**Pull request by chitianhao**
A coverity fix to the errors reported in the runroot functionality in traffic_layout. 
Modification in ``engine.cc`` and ``file_system.cc``.
And a small README document fix.

### CODE REVIEW

**Comment by dragon512** [comment]
[approve ci]

**Comment by dragon512** [comment]
[approve ci autest]

File: cmd/traffic_layout/engine.cc [hunk 1]
```diff
@@ -200,7 +200,10 @@ clean_parent(const std::string &bin_path)
 {
   char cwd[MAX_CWD_LEN];
   getcwd(cwd, sizeof(cwd));
-  std::string RealBinPath = realpath(bin_path.c_str(), nullptr); // bin path
+
+  char resolved_binpath[MAX_CWD_LEN];
+  realpath(bin_path.c_str(), resolved_binpath); // bin path
+  std::string RealBinPath = resolved_binpath;
 
   std::vector<std::string> TwoPath = {RealBinPath, cwd};
```
Comment (SolidWallOfCode) [inline cmd/traffic_layout/engine.cc]: Why isn't this `std::vector<ts::string_view>`?

File: cmd/traffic_layout/engine.cc [hunk 2]
```diff
@@ -200,7 +200,10 @@ clean_parent(const std::string &bin_path)
 {
   char cwd[MAX_CWD_LEN];
   getcwd(cwd, sizeof(cwd));
-  std::string RealBinPath = realpath(bin_path.c_str(), nullptr); // bin path
+
+  char resolved_binpath[MAX_CWD_LEN];
+  realpath(bin_path.c_str(), resolved_binpath); // bin path
```
Comment (SolidWallOfCode) [inline cmd/traffic_layout/engine.cc]: Shouldn't the return value of `realpath` be checked for `nullptr`?

File: cmd/traffic_layout/engine.cc (hunk 1 above)
Comment (dragon512) [reply cmd/traffic_layout/engine.cc]: because of memory ownership.

File: cmd/traffic_layout/engine.cc (hunk 2 above)
Comment (chitianhao) [reply cmd/traffic_layout/engine.cc]: If we have the second argument ``resolved_binpath`` defined as char [], there is no need to check return value. There will be no nullptr. I can add a line of checking ``strlen()`` of ``resolved_binpath`` value zero to print out errno.

**Pull request merged**